libdrgn/dwarf_index.c

```diff
@@ -492,6 +488,35 @@ struct drgn_error *drgn_dwarf_index_init(struct drgn_dwarf_index *dindex,
 	return NULL;
 }
 
+static void drgn_dwarf_index_cu_deinit(struct drgn_dwarf_index_cu *cu) {
+	free(cu->file_name_hashes);
+	free(cu->abbrev_insns);
+	free(cu->abbrev_decls);
+}
+
+static void drgn_dwarf_index_deinit_namespaces(struct drgn_dwarf_index_namespace *parent_ns) {
+	const uint64_t ns_tag = DW_TAG_namespace;
+	struct drgn_error *err;
+	struct drgn_dwarf_index_iterator it;
+	struct drgn_dwarf_index_die *die;
+
+
+	err = drgn_dwarf_index_iterator_init(&it, parent_ns, NULL, 0, &ns_tag, 1, false);
```

Comment:
I don't like exposing `resolve_ns` as part of the iterator interface. This can just walk the shards manually so we don't need to use the iterator.